docs/adr/0004-translations.md

```diff
@@ -0,0 +1,53 @@
+
+```
+# publish_document/published.yml
+en:
+  publish_document:
+    published:
+      reviewed:
+        title: Content has been published
+        body: |
+          ‘%{title}’ has been published on GOV.UK.
+
+          It may take 5 minutes to appear live.
+
+```
+
+Rails translations have a few special behaviours, such as pluralization, raw HTML, and variables. The `%{title}` string in the above is an example of a variable, which a developer will set to the title of the document being published.
+
+## Decision
+
+Although we could use translations to extract all of the strings in the application, in some cases we felt this wasn't necessary, or that a different method should be used. The following is a summary of the rules we currently use.
+
+   * **Link and button labels** are not extracted. We think link and button labels are unlikely to change, and extracting them made the application tests harder to read by obfuscating some of the crucial steps in the test with translation keys.
+   * **Big guidance** is extracted into it's own Markdown files and stored alongside the corresponding HTML page that shows it. For example, the guidance for creating a new document is stored in `app/views/new_document/guidance.md`.
+   * **Domain data** that's static is extracted into custom YAML files. This application has two static models (DocumentTypeSchema and SupertypeSchema) that encapsulate domain concepts where the data is part of the application. We think domain data - whether it's a backend setting or a label - should be defined in one place so it can change together. In this application, we've chosen to store the backing files for the two static models in `app/formats`.
+   * **Global strings** (states and validation messages) are extracted using translations. As these strings aren't page-specific, we put them at the top-level of the translation hierarchy (in `states.yml` and `validations.yml`).
+   * **All other strings** are extracted using translations, in a hierarchy that follows the structure of the `app/views` directory. For example, the above example relates to `app/views/publish_document/published.html.erb`.
+
+Every instance of a string in the tests has been replaced according to the above rules, such that the tests continue to pass when an extracted string is changed. **Link and button labels** are not replaced, as they are not extracted in the code.
+
+We also configured Rails to raise an exception when we hit a page where a translation is missing, as we don't test all of the translations; this is done by setting `config.action_view.raise_on_missing_translations = true` in `application.rb`.
+
```

Comment:
Might be an idea to add a section on testing and the various options that were considered. As that seems to be the controversial area and probably the bit most likely to trip up anyone new to the project. 

As I recall the options we had were:

- Register every phrase in i18n files and use only keys to those in our tests:
  - Pros: No doubts of what to put in locale config, changes to phrases can't change test outcome
  - Cons: Feature tests are very difficult to read
- Register every phrase in i18n files and use a mix of text and translation look ups in tests
  - Pros:  No doubts of what to put in locale config, Feature tests can be easy to read
  - Cons: Content designers can break tests with a content change, Unclear what text should be free text and what should be a translation look up in test.
- Register only phrases we expect a content designer to edit
  - Pros:  Feature tests can be easy to read, changes to phrases can't change test outcome
  - Cons: Inconsistency between what's in locale files and in views for a particular component

Reply:
Good point. I've added a section about this, trying to avoid my natural bias ;-).